File: MediCareWebSpa/src/app/DTOs/models/permission.ts
import { PermissionType } from "../../enums/permission-type";
import { RolePermission } from "./role-permission";

export class Permission {
  id: number;
  description: string;
  doctorOnly: boolean;
  patientOnly: boolean;
  permissionType: PermissionType;
  permissionRoles: RolePermission[] = [];

  constructor(
    id: number,
    description: string,
    doctorOnly: boolean,
    patientOnly: boolean,
    permissionType: PermissionType
  ) {
    this.id = id;
    this.description = description;
    this.doctorOnly = doctorOnly;
    this.patientOnly = patientOnly;
    this.permissionType = permissionType;
  }
}

File: MediCareWebApi/Models/Permission.cs
using System.ComponentModel.DataAnnotations;
using MediCare.Enums;

namespace MediCare.Models
{
	public class Permission : Entity
	{
		[Required]
		[MaxLength(256)]
		public string Description { get; set; }

		[Required]
		public bool DoctorOnly { get; set; }

		[Required]
		public bool PatientOnly { get; set; }

		[Required]
		public PermissionType PermissionType { get; set; }

		public ICollection<RolePermission> PermissionRoles { get; set; } = new List<RolePermission>();
	}
}
